lms/djangoapps/instructor/views/coupons.py

```diff
@@ -142,11 +156,14 @@ def get_coupon_info(request, course_id):  # pylint: disable=unused-argument
         return JsonResponse({
             'message': _("coupon with the coupon id ({coupon_id}) is already inactive").format(coupon_id=coupon_id)
         }, status=400)  # status code 400: Bad Request
-
+    expiry_date = 'Never Expires'
+    if coupon.expiration_date:
+        expiry_date = coupon.display_expiry_date
```

Comment:
the `display_expiry_date` is just a` @property ` and not an actual field.

When checking for expired coupons, we figured if the expiry date is Dec 31 midnight, then it's better to store Jan 1 2015 00:00 and make sure the current time is less than the stored date rather than storing Dec 31 2014 11:59:59.999999 etc.

We had to subtract 1 from the date for displaying the expiry date to show it as 31st Dec rather than the DB's value everywhere so we decided to make a `@property` to avoid having to subtract 1 everywhere. The 'Never Expires' logic, however, has been moved to the template.